### PR TITLE
Add router nokia-g240wb

### PR DESCRIPTION
**Pull request by elulcao**
- **What does this PR aim to accomplish?:**

Added Router Nokia G-240W-B guide 

- **How does this PR accomplish the above?:**

Added Router Nokia G-240W-B guide for IPv4 and IPv6 addresses 

- **What documentation changes (if any) are needed to support this PR?:**

Added docs/routers/nokia-G240WB.md

Screen Shot 2022-10-12 at 1 10 44 PM

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [ x ] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 /.idea
 /node_modules/
 /site/
+Pipfile
```
Comment (yubiuser) [inline .gitignore:7]: Why did you add this here?

**Review by yubiuser — COMMENTED** [review]
Thanks for your submission. Before reviewing your PR in detail, could you please crop/compress+crop your images?

File: .gitignore (hunk 1 above)
Comment (elulcao) [reply .gitignore]: It was used for my local Python env but not intended to merge, to be honest I forgot remove after my last commit. Removed from current PR.

Comment (elulcao) [reply .gitignore]: Images compressed and resized to a smallest size.

File: docs/routers/nokia-G240WB.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+> Note:
+There is no single way to do it right. Chose the one best suited for you.
+This guide is `IPv4` and `IPv6` compatible, but it is not mandatory to use `IPv6`.
+
+### Prerequisites
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+- `Pi-hole` has enabled `DHCP server`.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:20]: Why do you need Pi-hole to be the DHCP server?

File: docs/routers/nokia-G240WB.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- `Pi-hole` has enabled `IPv6 support (SLAAC + RA)`.
+
+For this example, we will be using the `192.168.1.100` and `2806:103e:1f:face::1681:100` addresses for the following steps. You will need to substitute your own addresses in the following steps.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:23]: Why do you recommend a global address here? I can change if your ISP assigned a different prefix We usually recommend a ULA address.

https://discourse.pi-hole.net/t/use-ipv6-ula-addresses-for-pi-hole/2127

Comment (elulcao) [reply docs/routers/nokia-G240WB.md:23]: Since this Router doesn't have a section for ULA (explicitly) I configured with IPV6 SLAAC however I realized after reading your comment that the Router offers ULA if the DHCP6 is enabled.

File: docs/routers/nokia-G240WB.md (hunk 2 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:20]: Good question, actually this is not necessary since the Router can serve as DHCP server by default.
Removed from guide.

File: docs/routers/nokia-G240WB.md (hunk 3 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md]: I restarted the router and the IPV6 prefix has been updated, however the communication is over the link-local IP.

File: docs/routers/nokia-G240WB.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+
+For this example, we will be using the `192.168.1.100` and `fe80::192:168:1:100` addresses for the following steps. You will need to substitute your own addresses in the following steps.
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.100` and `xxxx::192:168:1:100` respectively, this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
+
+!!! warning
+    Do not use `127.0.0.1#5335` or `::1#5335` as `DNS server` in the `Raspberry Pi` because it will cause a `DNS loop` as the `Pi-hole` will send the queries to the router which in turn will send them to `Pi-hole`.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:28]: What are you talking about here? Do you have another DNS server running at `127.0.0.1#5335`? Why should this cause a DNS loop?

Comment (elulcao) [reply docs/routers/nokia-G240WB.md:28]: This part is referring to an old configuration in my Raspi, where I was using unbound. I forgot to uninstall unbound and its configurations and hence I was understanding that somehow the Raspi was looping the request due to the `127.0.0.1#5335`. I started again the installation and this time the local IP#5335 is not an issue.

Comment (elulcao) [reply docs/routers/nokia-G240WB.md]: Removed the phrase. 
Also I renamed `wlan0` to `eth0` to make the guide homogeneous with current router guides.

File: docs/routers/nokia-G240WB.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+e.g. `inet 192.168.1.100  netmask 255.255.255.0` and `inet6 fe80::192:168:1:100 prefixlen 64`
+
+### IPv4 configuration
+
+Go to **Network** --> **LAN**
+
+1. Set `IPv4 address` to `192.168.1.254`
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:51]: I think there is no need to change the IPv4 address of the router. It might be `192.168.1.1` already by default.

**Review by yubiuser — COMMENTED** [review]
Looking at your screenshots, your DHCP pool ranges from `.50` - `.250`. However, your Pi's static address `.100` is within the pool. Your router won't know this address is already taken and can hand out this address to some other device.
It's better to give devices static addresses outside of the DHCP pool to avoid this conflict. You should set your Pi's address to something <`.50`

File: docs/routers/nokia-G240WB.md (hunk 5 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:51]: It depends of the ISP configuration, in my case 192.168.1.254 is the default GW. I was told that other routers (provided by same ISP are using 192.168.1.1). 
I mentioned that user can set 192.168.1.1 or 192.168.1.254 depending in their default configuration.

Comment (elulcao) [reply docs/routers/nokia-G240WB.md:51]: I configured the router to set static IP to the Raspi, the screenshot isn't including that section; please take a look to the pic attached to this message.

IMG_1275
* Im using wiregard so I can connect to home lan via LTE.

Comment (yubiuser) [reply docs/routers/nokia-G240WB.md:51]: This is a valid configuration. I think it's a bit cumbersome to set it static on the device and set a static DHCP assignment on the router, but your mileage may vary.  
However, for this public guide I would not recommend others to do so, because it's an additional step they need to perform. It's easier to separate static and dynamic IP pools.

Comment (elulcao) [reply docs/routers/nokia-G240WB.md:51]: I updated the IP address to x.x.1.10 for IPV4, with this change the IP address matches the range used by the DHCP IPV4.

Comment (elulcao) [reply docs/routers/nokia-G240WB.md]: Marked as "resolved", please let me know if this is OK to you. :)

File: docs/routers/nokia-G240WB.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+e.g. `inet 192.168.1.10  netmask 255.255.255.0` and `inet6 fe80::192:168:1:10 prefixlen 64`
+
+### IPv4 configuration
+
+Go to **Network** --> **LAN**
+
+1. Set `IPv4 address` to the default gateway address, `192.168.1.1` or `192.168.1.254` dependinding on your specific router configuration.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:51]:
```suggestion
1. Set `IPv4 address` to the default gateway address, `192.168.1.1` or `192.168.1.254` depending on your specific router configuration.
```

**Review by yubiuser — COMMENTED** [review]
Thanks. Now the screenshots don't fit the new IP of `192.168.1.10` anymore.

File: docs/routers/nokia-G240WB.md (hunk 6 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md]: That's a horrible typo!
The word is now correct: `depending`
The pics has been updated.

**Comment by elulcao** [comment]
> Thanks. Now the screenshots don't fit the new IP of `192.168.1.10` anymore.

Updated the pics, thanks for the feedback:)

**Comment by DL6ER** [comment]
@yubiuser You reviewed this before, do you still have any objections?

File: docs/routers/nokia-G240WB.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
+This guide was developed using a `Nokia G-240W-B` router. It should work for other models in the `G` series.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:1]:
```suggestion
This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
```

File: docs/routers/nokia-G240WB.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+### Device information
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:3]: This whole 'Device information' section can go. There is a lot of redundant and/or unnecessary information.

File: docs/routers/nokia-G240WB.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+- Device name: `Nokia G-240W-B`
+- Vendor: `Nokia`
+- Hardware version: `3FE56756ADBA`
+- Software version: `3FE47969AGCA21`
+- Chipset: `BCM6838`
+
+> Note:
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:11]:
```suggestion
!!! note
```

File: docs/routers/nokia-G240WB.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+There is no single way to do it right. Chose the one best suited for you.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:12]:
```suggestion
    There is no single way to do it right. Chose the one best suited for you.
```

File: docs/routers/nokia-G240WB.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+
+For this example, we will be using the `192.168.1.10` and `fe80::192:168:1:10` addresses for the following steps. You will need to substitute your own addresses in the following steps.
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively, this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:25]: This part should go into the section below (Grab your IPv4 and IPv6 address from your Raspberry Pi).

File: docs/routers/nokia-G240WB.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+There is no single way to do it right. Chose the one best suited for you.
+This guide is `IPv4` and `IPv6` compatible, but it is not mandatory to use `IPv6`.
+
+### Prerequisites
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+
+For this example, we will be using the `192.168.1.10` and `fe80::192:168:1:10` addresses for the following steps. You will need to substitute your own addresses in the following steps.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:21]: I think this part can be deleted. If users set static addresses, they should know them. If not, you provide a section below how to find them.

File: docs/routers/nokia-G240WB.md (hunk 8 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:3]: Got it, make sense if removing dup info.

File: docs/routers/nokia-G240WB.md (hunk 11 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:15]: Got you, updated the lines to the correct section.

File: docs/routers/nokia-G240WB.md (hunk 12 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md]: Sure thing, non used info must be removed; including the redundant text.

**Comment by elulcao** [comment]
@yubiuser thanks for the review, the guide now has the latest suggestion.

File: docs/routers/nokia-G240WB.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
+This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
+
+!!! note
+    There is no single way to do it right. Chose the one best suited for you.
+This guide is `IPv4` and `IPv6` compatible, but it is not mandatory to use `IPv6`.
+
+### Prerequisites
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:9]: Why did you remove the information about a static IP (or fixed DHCP assignment)?

File: docs/routers/nokia-G240WB.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively, this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:13]: I don't understand the second part of this sentence. What do you want to say here

`.... this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.`

File: docs/routers/nokia-G240WB.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+
+!!! note
+    There is no single way to do it right. Chose the one best suited for you.
+This guide is `IPv4` and `IPv6` compatible, but it is not mandatory to use `IPv6`.
+
+### Prerequisites
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:11]: Please re-order the paragraphs in this section. It makes sense to start with "SSH into...." followed by the expected output and then everything else.

File: docs/routers/nokia-G240WB.md (hunk 13 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:9]: I dont recall why, maybe I misunderstood your last comment. Adding back that section.

File: docs/routers/nokia-G240WB.md (hunk 14 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:13]: Removing that part since I don't want to confuse users.

File: docs/routers/nokia-G240WB.md (hunk 15 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:11]: Re-order the text, it must be visible in the next commit.

**Comment by elulcao** [comment]
@yubiuser I addressed the last moments, this time the text should be better. Let me know if you want me to do another update to the Readme file. 
Thanks.

File: docs/routers/nokia-G240WB.md [hunk 16]
```diff
@@ -0,0 +1,94 @@
+This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
+
+!!! note
+    There is no single way to do it right. Chose the one best suited for you.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:4]:
```suggestion
    There is no single way to do it right. Choose the one best suited for you.
```

File: docs/routers/nokia-G240WB.md [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+This guide is `IPv4` and `IPv6` compatible, but it is not mandatory to use `IPv6`.
+
+### Prerequisites
+
+- `SSH` into your `Raspberry Pi` and get the `IPv4` and `IPv6` address with `ifconfig eth0` (substituting `eth0` for the applicable network interface).
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:9]:
```suggestion

```

This is written blow in the section "Grab yozr IPv4...." already

File: docs/routers/nokia-G240WB.md [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:10]:
```suggestion
- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well. This can be a fixed DHCP assignment or configured statically on the Pi itself.
```

**Comment by elulcao** [comment]
Hi @yubiuser thanks for the latest comments, I didn't notice the typo. The duplicate entry, I noticed the same but I was unsure if remove or not since it was chained to our previous conversations

File: docs/routers/nokia-G240WB.md [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+This guide is `IPv4` and `IPv6` compatible, but it is not mandatory to use `IPv6`.
+
+### Prerequisites
+
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well. This can be a fixed DHCP assignment or configured statically on the Pi itself.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:14]:
```suggestion
```

File: docs/routers/nokia-G240WB.md [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
+
+SSH into your `Raspberry Pi`. A quick `ifconfig eth0` (substituting `eth0` for the applicable network interface) will get you the address information:
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:18]: I would put this section and the terminal output as first paragraph in "Grab your IPv4 and IPv6 address from your Raspberry Pi"

File: docs/routers/nokia-G240WB.md [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+### Prerequisites
+
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well. This can be a fixed DHCP assignment or configured statically on the Pi itself.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively.
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:16]: This could be the last paragraph in "Grab your IPv4 and IPv6 address from your Raspberry Pi"

File: docs/routers/nokia-G240WB.md [hunk 22]
```diff
@@ -0,0 +1,93 @@
...
+        inet6 fe80::e6f3:5490:577e:4fed  prefixlen 64  scopeid 0x20<link>
+        inet6 fe80::192:168:1:10  prefixlen 64  scopeid 0x20<link>
+        ether e4:5f:01:cd:8e:e1  txqueuelen 1000  (Ethernet)
+        RX packets 4342  bytes 704995 (688.4 KiB)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 3376  bytes 661707 (646.1 KiB)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+e.g. `inet 192.168.1.10  netmask 255.255.255.0` and `inet6 fe80::192:168:1:10 prefixlen 64`
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:34]:
```suggestion
The relevant lines here are `inet 192.168.1.10  netmask 255.255.255.0` and `inet6 fe80::192:168:1:10 prefixlen 64`
```

**Review by yubiuser — COMMENTED** [review]
As (hopefully) last remark: I would re-order the structure of "Grab your IPv4 and IPv6 address from your Raspberry Pi"

And please squash down all your commits into a singel one.

File: docs/routers/nokia-G240WB.md (hunk 20 above)
Comment (elulcao) [reply docs/routers/nokia-G240WB.md:14]: Addressed the comments for this section.

**Comment by elulcao** [comment]
> As (hopefully) last remark: I would re-order the structure of "Grab your IPv4 and IPv6 address from your Raspberry Pi"
> 
> And please squash down all your commits into a singel one.

Hello @yubiuser, hope your doing good!
I finally squashed and addressed the comments for this PR 👍 Could you share an +1?
Let me know if anything else is needed.

File: docs/routers/nokia-G240WB.md [hunk 23]
```diff
@@ -0,0 +1,91 @@
+This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
+
+!!! note
+There is no single way to do it right. Choose the one best suited for you.
```
Comment (yubiuser) [inline docs/routers/nokia-G240WB.md:4]: I think here is some indention missing, as the 'note' is only an empty box. See the preview: https://deploy-preview-791--pihole-docs.netlify.app/routers/nokia-G240WB/

**Review by yubiuser — APPROVED** [review]
